### PR TITLE
Remove javaagent-all

### PR DESCRIPTION
**Pull request by laurit**
`splunk-otel-javaagent.jar` and `splunk-otel-javaagent-all.jar` are the same.

### CODE REVIEW

**Review by theletterf — APPROVED** [review]

**Comment by theletterf** [comment]
@mateuszrzeszutek @laurit This will require a docs and GDI UI edit, as we're currently linking to the `-all` version for some reason. I'm concerned there might be integrations or scripts out there that use it. I recommend keeping both files for now until we know about usage (if possible).

**Comment by mateuszrzeszutek** [comment]
Let's just switch to `splunk-otel-javaagent.jar` without removing `splunk-otel-javaagent-all.jar` -- we can deprecate the latter in this release and remove it later. WDYT?

**Comment by theletterf** [comment]
Fine by me—the change in the docs is relatively trivial. It's the scripts and automations I'm more concerned about, though there's no way of getting around that breaking change. Could we run a search in support tickets to peek usage?

**Comment by mateuszrzeszutek** [comment]
> Could we run a search in support tickets to peek usage?

No idea 🤷  Probably we can, but I've no idea if anything will turn up - and if something will, it probably won't be all of -all usages anyway.
It's probably best to deprecate this and just wait for a couple of releases.

File: Dockerfile [hunk 1]
```diff
@@ -1,3 +1,3 @@
 FROM busybox 
 
-ADD dist/splunk-otel-javaagent-all.jar /
+ADD dist/splunk-otel-javaagent.jar /
```
Comment (mateuszrzeszutek) [inline Dockerfile:3]: @owais FYI

**Comment by laurit** [comment]
> Let's just switch to `splunk-otel-javaagent.jar` without removing `splunk-otel-javaagent-all.jar` -- we can deprecate the latter in this release and remove it later. WDYT?

I reverted all non doc changes besides cloudfoundlry buildpack where just the download url changes

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**